### ISSUE TITLE
Invalid configuration mapping when import Microsoft.CognitiveServices.Speech nuget

### ISSUE THREAD

**Issue by ductranit** [Azure-Samples/cognitive-services-speech-sdk#762]
I try to import speech sdk into my azure function in visual studio for mac.
I don't see any error in the nuget package console. But it has an `Invalid configuration mapping` error in the project tree.
Here is my csproj:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.0.0" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.3" />
    <PackageReference Include="Microsoft.CognitiveServices.Speech" Version="1.13.0" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <Folder Include="Shared\" />
    <Folder Include="Shared\Services\" />
  </ItemGroup>
</Project>
```
I searched around and tried [this solution](https://developercommunity.visualstudio.com/content/problem/443746/new-project-configuration-added-on-vs2017-on-windo.html) to add platform configs
```
<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug_macOS|AnyCPU' " />   
 <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release_macOS|AnyCPU' " />
```
but it still didn't work.
The project can be built and run, but it can't find any azure http function.
```
 No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
```

If I remove the nuget `Microsoft.CognitiveServices.Speech` then it works fine. 
Isn't the speech sdk compatible with .netcore or azure function?

**Version of the Cognitive Services Speech SDK**
1.13.0

**Platform, Operating System, and Programming Language**
 - OS: macos
 - Hardware - x64
 - Programming language: C#

**Comment by glecaros** [comment]
Hi @ductranit, can you please share a min repro project we can use to investigate this?

**Comment by ductranit** [comment]
@glecaros  Here is it
[TestAzureSpeech.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/5060351/TestAzureSpeech.zip)
I just create a new function project from the template and add package `Microsoft.CognitiveServices.Speech`

**Comment by glecaros** [comment]
I see, I'll try to dig on what's happening. Thanks for the details!

**Comment by ductranit** [comment]
Updated: If I run the project in visual studio code on mac, then I can see the azure function is loaded, but it has this exception when access the speech service:

```
Microsoft.CognitiveServices.Speech.csharp: Unable to load shared library 'libMicrosoft.CognitiveServices.Speech.core.dylib' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(liblibMicrosoft.CognitiveServices.Speech.core.dylib.dylib, 1): image not found.
``

**Comment by jhakulin** [comment]
@ductranit Thanks for details, this is issue in our SDK for Mac, we are working on to fix this.

**Comment by pankopon** [comment]
Closing this, as we have added a work item to our backlog and will address it in the next or later release. Once the work is done, we will update this issue to let you know what release the fix will be in.

**Comment by jhakulin** [comment]
@ductranit There is possible issue with Visual Studio for Mac related to speech SDK nuget. I will continue to follow-up issue with them.

I tested with Visual Studio Code for Mac and with basic HttpTrigger Azure function template and added Microsoft.CognitiveServices.Speech 1.13.0 nuget and code which uses speech SDK to Azure function. Then I run the test using localhost and tested inside Azure portal and both sides were functioning correctly. I was following instructions from here:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-vs-code?pivots=programming-language-csharp, I had latest Visual Studio Code for Mac with Azure extensions.

**Comment by ductranit** [comment]
@jhakulin I come back to here after a year, it's so amazing that I still see the same issue on latest vs for mac. 
I also tried with vscode and got the same.
```
Microsoft.CognitiveServices.Speech.csharp: Unable to load shared library 'libMicrosoft.CognitiveServices.Speech.core.dylib' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(liblibMicrosoft.CognitiveServices.Speech.core.dylib.dylib, 1): image not found.
```
I'm using the latest package `Microsoft.CognitiveServices.Speech` `1.18.0`

**Comment by jhakulin** [comment]
@ductranit I just tried these samples with Visual Studio for Mac version 8.10.7 (build 17)
https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnetcore/from-microphone
https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnetcore/text-to-speech

And I feel your pain with first experience of running those samples, they failed to .NET failure with me which was not clear from the Visual Studio error (ie. I did not see the speech SDK error as you reported there). After that I rebooted the Mac and tried the samples again and now all work.
 
![image](https://user-images.githubusercontent.com/16473503/129762892-50e88184-831d-4eae-9e43-548a40afa992.png)

So there is some instability experienced with VS on Mac. I can report these issues to VS team, especially if I can reproduce those again. As crazy as it sounds, at the moment, could you try if reboot helps?

The original error you reported "invalid configuration mapping" seems to be fixed in VS on Mac now.

**Comment by ductranit** [comment]
@jhakulin yes, I can run the github sample on my mac and I confirm that `invalid configuration mapping` issue is fixed
Seem this issue only happen on azure function when run on macos (it's fine when I debug on windows)
From above comment I gave you the `TestAzureSpeech.zip`  azure function sample  but you may not notice it.
I update the reproduce code with latest sdk version here:
[TestAzureSpeech.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/7001656/TestAzureSpeech.zip)
Can you run it on your mac?

**Comment by jhakulin** [comment]
@ductranit You are correct that with AzureFunction there is still some problems. When I created simple http trigger and added Speech SDK to that, I get the following error:

[8/17/2021 10:06:14 PM] System.Private.CoreLib: Exception while executing function: JarnosTrigger. Microsoft.CognitiveServices.Speech.csharp: 
Unable to load shared library 'Microsoft.CognitiveServices.Speech.core.dll' or one of its dependencies. In order to help diagnose loading problems, 
consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libMicrosoft.CognitiveServices.Speech.core.dll, 1): image not found.

Since error is pointing to Microsoft.CognitiveServices.Speech.core.dll, not Mac dynamic library at all, I think the problem is some kind of configuration problem with Visual Studio on Mac not being able to find the correct dynamic library from the nuget. I'll check with VS team.

Earlier as I commented here (https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/762#issuecomment-682098570) VS code was working with Azure functions on Mac, did not test that again now.
